BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -297,23 +298,32 @@ - (BOOL)isFinished
 
 - (void)start
 {
-    [[BOXAPIOperation APIOperationGlobalLock] lock];
+    if (self.shouldStartImmediately) {
+        if ([self isReady]) {
+            self.state = BOXAPIOperationStateExecuting;
+            [self executeOperation];
+        } else {
+            BOXAssertFail(@"Operation was not ready but start was called");
+        }
+    } else {
+        [[BOXAPIOperation APIOperationGlobalLock] lock];
 
-    if ([self isReady])
-    {
-        // Set state = executing once we have the lock
-        // BOXAPIQueueManagers check to ensure that operations are not executing when
-        // they grab the lock and are adding dependencies.
-        self.state = BOXAPIOperationStateExecuting;
+        if ([self isReady])
+        {
+            // Set state = executing once we have the lock
+            // BOXAPIQueueManagers check to ensure that operations are not executing when
+            // they grab the lock and are adding dependencies.
+            self.state = BOXAPIOperationStateExecuting;
 
-        [self performSelector:@selector(executeOperation) onThread:[[self class] globalAPIOperationNetworkThread] withObject:nil waitUntilDone:NO];
-    }
-    else
-    {
-        BOXAssertFail(@"Operation was not ready but start was called");
-    }
+            [self performSelector:@selector(executeOperation) onThread:[[self class] globalAPIOperationNetworkThread] withObject:nil waitUntilDone:NO];
```

Comment:
I'm curious what `globalAPIOperationNetworkThread` gives us. I'm pretty sure it's a thread because it was written in a time before GCD. It's probably used for serialization, that we may or may not need for correctness. If we're sure it's safe to skip it, a comment about why might be useful.

Reply:
I'm not sure if there are other intention for globalAPIOperationNetworkThread besides it's done before GCD time. The thread entry point does wrap the runloop inside autoreleasepool, so we'll need to make sure we add autoreleasepool around operation execution. Updated